src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialProvider.java

```diff
@@ -80,8 +87,9 @@ public void startWatchingForSecrets() {
             Config config = cb.build();
             DefaultKubernetesClient _client = new DefaultKubernetesClient(config);
             LOG.log(Level.FINER, "Using namespace: {0}", _client.getNamespace());
-            LOG.log(Level.FINER, "retreiving secrets");
-            SecretList list = _client.secrets().withLabel(SecretUtils.JENKINS_IO_CREDENTIALS_TYPE_LABEL).list();
+            LOG.log(Level.INFO, "retrieving secrets with selector: {0}, {1}", new String[]{SecretUtils.JENKINS_IO_CREDENTIALS_TYPE_LABEL, labelSelector});
+            LabelSelector selector = LabelSelectorExpressions.parse(labelSelector);
```

Comment:
this should be producing a spotbugs warning as selector may be passed null and the default is non null.

for some reason it appears as though it is not :-(